src/internal.c

```diff
@@ -16512,9 +16512,55 @@ void PickHashSigAlgo(WOLFSSL* ssl, const byte* hashSigAlgo,
             ssl->suites->hashAlgo = sha512_mac;
             break;
         }
+    #endif
+    #ifdef HAVE_ECC
```

Comment:
This changes the sig/hash algo selection.  How do we know, there will be no interop issues?

Reply:
That's a good point. Previous to these changes we could choose a hash algo like SHA512 for a P-256 ephemeral key. My preference would be to default to this new behavior and have a build option to disable this new section of code. Its my opinion this new code is more compatible and less likely to have issues. @toddouska : What is your preference?

Reply:
I'd like to know lots and lots and lots of interop testing was done.  Ideally this change would be in a separate commit in a separate PR because there's a chance it will cause issues.